src/kvstore/Listener.cpp

```diff
@@ -251,5 +252,17 @@ std::pair<int64_t, int64_t> Listener::commitSnapshot(const std::vector<std::stri
     return std::make_pair(count, size);
 }
 
+void Listener::resetListener() {
+    std::lock_guard<std::mutex> guard(raftLock_);
+    committedLogId_ = proposedTerm_ = lastLogTerm_ = term_ = lastApplyLogId_ = 0;
+    lastTerm_ = lastId_ = -1;
+    wal_->reset();
+    persist(lastApplyLogId_, lastTerm_, lastApplyLogId_);
+}
+
+bool Listener::rebuildDone(LogID baseline) {
+    return lastApplyLogId_ >= baseline * FLAGS_ft_rebuild_threshold ? true : false;
```

Comment:
Could consider compare `lastApplyLogId_` with `lastLogId_` in RaftPart. I see you get `lastApplyLogId_` in the task, and pass as baseline, it could be 0 if listener is newly added.

Reply:
> Could consider compare `lastApplyLogId_` with `lastLogId_` in RaftPart. I see you get `lastApplyLogId_` in the task, and pass as baseline, it could be 0 if listener is newly added.

It could be the same problem，`lastApplyLogId_` and `lastLogId_` will be set to 0 when resetListener.  It causes the same problem.  for current, if the baseline is 0, means it doesn't need Rebuild and should end immediately.Interesting. Let's discuss it offline.